packages/csharp/src/components/ClassMethod.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  Block,
+  Children,
+  MemberDeclaration,
+  refkey,
+  Refkey,
+  Scope,
+} from "@alloy-js/core";
+import {
+  AccessModifier,
+  computeModifiersPrefix,
+  getAccessModifier,
+  getAsyncModifier,
+  getMethodModifier,
+  MethodModifier,
+} from "../modifiers.js";
+import { useCSharpNamePolicy } from "../name-policy.js";
+import { CSharpOutputSymbol } from "../symbols/csharp-output-symbol.js";
+import { CSharpMemberScope, useCSharpScope } from "../symbols/scopes.js";
+import { ParameterProps, Parameters } from "./Parameters.jsx";
+
+// properties for creating a method
+export interface ClassMethodProps {
+  name: string;
+  refkey?: Refkey;
+  children?: Children;
+  accessModifier?: AccessModifier;
```

Comment:
What do you think about following a bool prop pattern for accessModifiers similar to the TypeScript compoentns? I like how ergonomic it is to just say <ClassMethod public static ... /> rather than <ClassMethod accessModifiers=["public"] />

Not in the scope of this PR but something to think about.

Reply:
I think we can have better docs this way, like we can document the `async` bool prop and say if you are declaring an async `void` there is no need to pass the return prop...

Reply:
oh wait, async is already a bool prop

Reply:
+1 bool props is the way.